app/code/Magento/Config/Model/Config/Structure.php

```diff
@@ -191,7 +191,7 @@ public function getElementByConfigPath($path)
     {
         $allPaths = $this->getFieldPaths();
 
-        if (isset($allPaths[$path])) {
+        if (isset($allPaths[$path]) && is_array($allPaths[$path])) {
```

Comment:
Could you explain why it wasn't needed earlier?

Reply:
We didn't call it with empty $allPaths before and with the changes at the 'validate' method we can have an error at $path = array_shift($allPaths[$path]) because '$allPaths[$path]' could be 0 (int) instead of array.